packages/wrangler/src/pages/publish.tsx

```diff
@@ -250,7 +250,7 @@ export const Handler = async ({
 		commitMessage,
 		commitHash,
 		commitDirty,
-		bundle: bundle ?? !noBundle,
+		bundle: bundle || !noBundle,
```

Comment:
If I'm reading this correctly, I think we now have:

| | `--bundle` undefined | `--bundle=false` | `--bundle=true` |
| --- | --- | --- | --- |
| `--no-bundle` undefined | `false` | `false` | `true` |
| `--no-bundle` false | `true` | `true` | `true` |
| `--no-bundle` true | `false` | `false` | `true` |

Which still doesn't quite sound correct to me. I think it should be:

| | `--bundle` undefined | `--bundle=false` | `--bundle=true` |
| --- | --- | --- | --- |
| `--no-bundle` undefined | `false` | `false` | `true` |
| `--no-bundle` false | `true` | **ERROR** | `true` |
| `--no-bundle` true | `false` | `false` | **ERROR** |

And eventually, (not in this PR, but soon), we'll flip the defaults so it'll become:

| | `--bundle` undefined | `--bundle=false` | `--bundle=true` |
| --- | --- | --- | --- |
| `--no-bundle` undefined | **`true`** | `false` | `true` |
| `--no-bundle` false | `true` | ERROR | `true` |
| `--no-bundle` true | `false` | `false` | ERROR |

Reply:
I think this should also be changed here: https://github.com/cloudflare/wrangler2/blob/416babf050ed3608e0fd747111561a4c7207185e/packages/wrangler/src/pages/dev.ts#L279

And then we should check that Wrangler proper (for both `dev` and `publish`) also handles the `--bundle` and `--no-bundle` args in a good way.

Reply:
ah ok so you're saying that you want me to throw errors when the two flags are in conflict? yeah that makes sense 😅

(by the way, the case in which both are `undefined` I think that the result is `true` not false, since `!undefined` is `true`)

Reply:
Ah, you're right. But it should be `false` for now. We want bundling to _default_ to `false` in Pages right now while it's new. We'll flip it to default `true` soon.

Reply:
No, I was correct the first time, actually. We have default values up higher in this file that means we're defaulting to `false` if both `--no-bundle` (default `true`) and `--bundle` (default `false`) are unset.

Eventually we'll flip those defaults like resulting in that final third table. I've updated the tables back again to what they were originally.

Reply:
yeah you're right, when the are both `undefined` it should be `false` as you can see in this test here: https://github.com/cloudflare/wrangler2/pull/2609/files#r1086508479

sorry about the confusion 😓